### PR TITLE
Add support for all EVPN bundle services

### PR DESCRIPTION
**Pull request by ipspace**
* Add 'evpn.bundle' VRF parameter
* Validate 'evpn.bundle' value against evpn.attributes.bundle list
* Check 'evpn.bundle' value against device.features.evpn.bundle list
* Refactor VLAN bundle code in EVPN configuration module
* Fix EVPN VLAN bundle integration test case
* Add 'evpn.bundle' to EVPN asymmetric IRB test case
* Update the documentation

Implements #636

### CODE REVIEW

File: netsim/modules/evpn.py [hunk 1]
```diff
@@ -114,6 +114,55 @@ def vlan_aware_bundle_service(vlan: Box, vname: str, topology: Box) -> None:
...
+"""
+Create VLAN services -- either as independent VLANs or as a VRF-based bundle
+"""
+
+def create_vlan_service(vname: str, topology: Box) -> None:
+  vlan = topology.vlans[vname]
+  if not get_vlan_bundle_flag(vlan,vname,topology):
+    vlan_based_service(vlan,vname,topology)
+  else:
+    vlan_aware_bundle_service(vlan,vname,topology)
```
Comment (jbemmel) [inline netsim/modules/evpn.py:148]: This method is now a bit of a misnomer - ```vlan_bundle_service``` would be more accurate

File: netsim/modules/evpn.py [hunk 2]
```diff
@@ -114,6 +114,55 @@ def vlan_aware_bundle_service(vlan: Box, vname: str, topology: Box) -> None:
...
+
+"""
+Validate that there's no evpn.bundle setting on node VRFs
+"""
+
+def validate_no_node_vrf_bundle(node: Box, topology: Box) -> None:
```
Comment (jbemmel) [inline netsim/modules/evpn.py:154]: Is this common to have attributes that are only valid on global vrfs, but not local vrfs? Do we want to distinguish between the two in the data model?

**Review by jbemmel — APPROVED** [review]

File: netsim/modules/evpn.py (hunk 1 above)
Comment (ipspace) [reply netsim/modules/evpn.py:148]: In theory, you're correct. In practice, let's wait and see whether we need anything more than what this one contains (when you implement other bundle services).

File: netsim/modules/evpn.py (hunk 2 above)
Comment (ipspace) [reply netsim/modules/evpn.py:154]: Definitely evpn.transit_vni, maybe also RD/RT. Not sure it's worth modifying common code to cope with them (at the moment).

**Comment by ipspace** [comment]
Thanks for the review and the comments! Merging...

**Pull request merged**